issue_manager_script.py

```diff
@@ -5,83 +5,49 @@
 from github.GithubException import GithubException, RateLimitExceededException
 
-# Edge Case 1: Rate Limiting
-# - Monitor rate limits and handle RateLimitExceededException.
-
-# Edge Case 2: Time Zone and Timestamp Issues
-# - Ensure consistent use of UTC and add a buffer to 'since_time'.
-
-# Edge Case 3: Duplicate Issues in the Project
-# - Improve duplicate checking logic.
-
-# Edge Case 5: Repositories with Issues Disabled
-# - Skip repositories where issues are disabled.
-
-# Edge Case 6: Archived or Disabled Repositories
-# - Skip archived repositories.
-
-# Edge Case 8: Renamed or Deleted Repositories
-# - Handle repositories that have been renamed or deleted.
-
-# Edge Case 9: Special Characters in Repository Names
-# - Sanitize repository names when creating columns.
-
-# Edge Case 10: Issues Transferred Between Repositories
-# - Handle issues moved between repositories.
-
-# Edge Case 11: Handling Closed or Reopened Issues
-# - Optionally remove closed issues from the project.
-
-# Edge Case 12: Issues Created Before the Last Run
-# - Add a buffer to 'since_time' to prevent missing issues.
-
-# Edge Case 14: Handling Pull Requests
-# - Exclude pull requests if desired.
-
-# Edge Case 16: Concurrency and Race Conditions
-# - Prevent overlapping runs with a lock file.
-
-# Edge Case 17: Manual Modifications to the Project
-# - Respect manual changes by users.
-
-# Edge Case 18: Error Handling and Notifications
-# - Implement robust error handling and notifications.
-
-# Edge Case 21: Repository Forks
-# - Skip forked repositories.
-
-# Edge Case 22: Reaching Maximum API Page Limits
-# - Implement proper pagination.
-
-# Edge Case 24: Exception Handling in Loops
-# - Add try-except blocks around API calls.
-
-# Edge Case 25: Changes in Repository Ownership
-# - Handle repositories transferred out of the organization.
+# --------------------------------- Overview ----------------------------------
+# This script automates the process of adding new issues from any repository
+# within the 'WhyDRS' organization to a specified GitHub Project. It handles
+# various edge cases to ensure robustness, such as rate limiting, time zone
+# discrepancies, duplicate issues, and more.
+
+# The script performs the following main tasks:
+# 1. Authenticates with GitHub using a GitHub App.
+# 2. Fetches the specified organization and project.
+# 3. Iterates over all repositories in the organization.
+# 4. For each repository, it processes new issues created within the last day.
+# 5. Adds new issues to the project under columns named after the repository.
+# 6. Handles edge cases like archived repositories, forked repositories,
+#    rate limiting, and more.
+
+# --------------------------- Configuration Variables -------------------------
+
+# Environment variables for GitHub App credentials
+app_id = os.environ['APP_ID']                # GitHub App ID
+private_key = os.environ['APP_PRIVATE_KEY']  # GitHub App private key in PEM format
 
```

Comment:
I've been thinking more on this since the [original security chat](https://github.com/WhyDRS/.github/issues/5) just in terms of general admin. Perhaps we can talk tonight about specifics. In general, I love the approach you're using here to get things up and running using the internal Projects functionality. 🚀

In the future, could it be helpful to use more public "data-scraping" tools like the GitHub webhook, since everything's open for external review anyway?

Reply:
Love the consideration for security here. Reading up on Github webhooks, it seems that they need an external server or cloud function to have set up (correct me if I'm wrong here). Webhooks also are limited to one repository when combined with GH Actions; setting up a template to duplicate across multiple repositories could work, but would require more maintenance and attention when new repositories are created. I know Personal Access Tokens were also discussed before, and doing some research I found that long-term tokens are less secure than ephemeral  tokens. What's nice about this GH app is that it generates a new token when it's run that lasts an hour and then expires. As far as permissions that the app has, it's only configured for Read & Write permissions solely for Issues and Projects within the organization (I like the granularity aspect of this). Securing the app includes securing the private keys that were generated for it, which I have done using AES-128 encryption on a folder level on my laptop. I think it would be good practice to have a standard for this written down in a security framework [as mentioned here](https://github.com/WhyDRS/.github/issues/5).

Reply:
Makes good sense to formalize the security framework - and I also like the idea of the tokens expiring, which is every bit as loggable but more secure. Would the transition to the single landing page framework for issues across repos in the org allow for the webhook to support this? I also was curious for clarity - does this security token get passed on in some way to the user or submitter or does it stay appside to allow for the permissions in the moment when new items are added?

Reply:
For a separate landing page, GH apps + actions can be used to fetch issues. This same strategy can also be used to send messages to Discord without using webhooks. The ephemeral security tokens generated within a GH app are used exclusively by the app itself to authenticate and authorize actions on Github, so they stay on the application side. If you had another use case for webhooks in mind please let me know.

Reply:
This is some stellar security discussion from both of you! 💜 Thanks so much for diving deep into these nuances. 🧠 Marking as resolved for the scope of this review, given merged, but always can reference in further Discord growth chats! 🚀